osu.Game/Overlays/Mods/ModSelectScreen.cs

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Framework.Layout;
+using osu.Game.Configuration;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets.Mods;
+using osuTK;
+using osuTK.Input;
+
+namespace osu.Game.Overlays.Mods
+{
+    public class ModSelectScreen : OsuFocusedOverlayContainer
```

Comment:
Naming is a touch weird. A screen which is actually an overlay. Was this mostly to avoid conflicts with the existing class?

Reply:
Correct, this was to avoid `ModSelectOverlayV2` or similar atrocities. I figured it was fine to call this a "screen" since it's more... "screeny" than the old overlay, given that it takes up the entire screen (bar toolbar, maybe). Open to suggestions for better names, or to moving these names back to `...Overlay` when this entire ordeal is done.